app/src/common/shared/com/igalia/wolvic/ui/widgets/Windows.java

```diff
@@ -163,15 +163,20 @@ class WindowsState {
     private DownloadsManager mDownloadsManager;
     private ConnectivityReceiver mConnectivityReceived;
 
-    @IntDef(value = {NONE, BOOKMARKS, WEB_APPS, HISTORY, DOWNLOADS, ADDONS, NOTIFICATIONS})
-    public @interface PanelType {}
-    public static final int NONE = 0;
-    public static final int BOOKMARKS = 1;
-    public static final int WEB_APPS = 2;
-    public static final int HISTORY = 3;
-    public static final int DOWNLOADS = 4;
-    public static final int ADDONS = 5;
-    public static final int NOTIFICATIONS = 6;
+    public enum ContentType {
```

Comment:
The code tend to use annotations instead of enums because dev docs were not recommending them because they become classes and thus increase the memory footprint. It's explained in sources like https://medium.com/android-news/android-performance-avoid-using-enum-on-android-326be0794dc3. Although recently I've read that proguard optimizes them so maybe it isn't a big deal anymore.

Reply:
Apparently this page used to have a suggestion to avoid enums (it is linked in old StackOverflow threads) but it does not have that warning anymore https://developer.android.com/topic/performance/memory

As I understand it, enums still use more memory than integer constants but nowadays the difference is not large.

Reply:
fair enough